upstairs/src/lib.rs

```diff
@@ -834,6 +834,40 @@ async fn cmd_loop(
     let mut ping_interval = deadline_secs(10);
     let mut timeout_deadline = deadline_secs(50);
 
+    let (tx, mut rx) = mpsc::channel(100);
+
+    {
+        let up_c = up.clone();
+        let up_coms_c = up_coms.clone();
+
+        tokio::spawn(async move {
+            loop {
+                tokio::select! {
```

Comment:
Maybe make this a while on the rx.recv() like we did for the downstairs.
We don't need the select or the loop then, so it's two less indents.  Just think of all
the whitespace this change could save.

Reply:
done in c6fe5c96b929c1288bd2f8c7c87d0b64c8d893fb